financial/present_value.py

```diff
@@ -0,0 +1,37 @@
+# Reference: https://www.investopedia.com/terms/p/presentvalue.asp
+
+# Algorithm that calculates the present value of a stream of yearly cash flows given...
+# 1. The discount rate (as a decimal, not a percent)
+# 2. An array of cash flows, with the index of the cash flow being the associated year
+
+# Note: This algorithm assumes that cash flows are paid at the end of the specified year
+
+
+def present_value(discount_rate: float, cash_flows: list[float]) -> float:
+    """
+    >>> round(present_value(0.13, [10, 20.70, -293, 297]), 2)
+    4.69
+    >>> round(present_value(0.07, [-109129.39, 30923.23, 15098.93, 29734,39]), 2)
+    -42739.63
+    >>> round(present_value(0.07, [109129.39, 30923.23, 15098.93, 29734,39]), 2)
+    175519.15
+    >>> present_value(-1, [109129.39, 30923.23, 15098.93, 29734,39])
+    Traceback (most recent call last):
+        ...
+    ValueError: Invalid discount rate, please choose a rate other than -1
+    """
```

Comment:
There are a few improvements that can be made to the `present_value` function to make it more robust and avoid potential errors:

1. Validate that the cash flows list is not empty before calculating the present value.

       if not cash_flows:
            raise ValueError("Cash flows list cannot be empty")

2. Validate that the discount rate is not negative, as negative discount rates don't make sense in finance.

       if discount_rate < 0:
        raise ValueError("Discount rate cannot be negative")

Reply:
Thank you for these improvements, I have updated the code.